pilot/pkg/model/sidecar.go

```diff
@@ -766,6 +763,38 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 		}
 	}
 
+	if features.UnifiedSidecarScoping {
+		type namespaceProvider struct {
+			Namespace         string
+			KubernetesService bool
+		}
+		validServices := make(map[host.Name]namespaceProvider, len(importedServices))
+		for _, svc := range importedServices {
+			np := namespaceProvider{Namespace: svc.Attributes.Namespace, KubernetesService: svc.Attributes.ServiceRegistry == provider.Kubernetes}
+			ex, f := validServices[svc.Hostname]
+			// Select a single namespace for a given hostname.
+			// If the same hostname is imported from multiple namespaces, pick the one in the configNamespace
+			// If neither are in configNamespace, an arbitrary one will be chosen, preferring Kubernetes services over ServiceEntry
+			if !f {
+				// First time we saw the hostname, take it
+				validServices[svc.Hostname] = np
+			} else if svc.Attributes.Namespace == configNamespace {
+				// We have seen the hostname, but now its in the configNamespace which has precedence
+				validServices[svc.Hostname] = np
```

Comment:
This is not right to me. Say two services for "foo.bar", both in confignamespace. The later will win, but we expect the older one win.

Should consider kubernetes service as well.  Not the case third branch below does.

Reply:
this is not picking a specific service! it's picking the namespace. this code already exists it's just been refactored/moved

Can you give a real example config that behaves correctly before this PR and incorrectly after this PR? Otherwise I have a very hard way to make progress

Reply:
I get your point. this is now selecting namespace actually.